test/testdata/namer/has_attached_class_subclass_class.rb

```diff
@@ -0,0 +1,11 @@
+# typed: true
+# disable-fast-path: true
+
+class A < Class
```

Comment:
Okay, just to make sure I understand it correctly, is the following what is happening?
- If the user subclasses `Class`, but does not call `has_attached_class!`, there will be a missing type member in the child class, so `reportRedeclarationError` will be called and we can report this error.
- If the user subclasses `Class`, but does call `has_attached_class!`, then there's no missing type member, so `reportRedeclarationError` is skipped, and we can't report this error, and instead report the namer error.

Reply:
* All of this should be exceedingly rare, because `class A < Class` isn't valid at runtime anyways and isn't something I've ever see someone try to do, so the only real question here is "how bad are the error messages relative to how common we expect this to be"